compiler/stable_mir/src/mir/body.rs

```diff
@@ -899,11 +899,9 @@ pub enum BorrowKind {
     /// Data must be immutable and is aliasable.
     Shared,
 
-    /// The immediately borrowed place must be immutable, but projections from
-    /// it don't need to be. This is used to prevent match guards from replacing
-    /// the scrutinee. For example, a fake borrow of `a.b` doesn't
-    /// conflict with a mutable borrow of `a.b.c`.
-    Fake,
+    /// An immutable, aliasable borrow that is discarded after borrow-checking. Can behave either
+    /// like a normal shared borrow or like a special shallow borrow (see [`FakeBorrowKind`]).
+    Fake(FakeBorrowKind),
```

Comment:
I don't know exactly how stable mir is used, but what I'd imagine is that we would only have `Shared`/`ShallowShared`/`Mutable` kinds of borrows, since "fake" only means "discard after borrowck". But this might trip up some users of stable mir

Reply:
Well I guess stable_mir has `FakeRead`, so it's consistent to have fake borrows too.